### ISSUE TITLE
Invalid utf-8

### ISSUE THREAD

**Issue by thewh1teagle** [tazz4843/whisper-rs#115]
![image](https://github.com/tazz4843/whisper-rs/assets/61390950/0a3b7424-4b6f-4fa7-8899-7d46b83caa02)
The song in hebrew:
[muminim hebrew.zip](https://github.com/tazz4843/whisper-rs/files/13896635/muminim.hebrew.zip)

**Comment by tazz4843** [comment]
This is an upstream issue, not something we can control. I run into this myself with my own services, and I just log it and ignore the output.

Doing some digging I found the following:
https://github.com/ggerganov/whisper.cpp/issues/1098
https://github.com/ggerganov/whisper.cpp/pull/1118

**Comment by thewh1teagle** [comment]
@tazz4843 
How can I ignore the errors and take only some of the transcribed data? or if it's in some languages it won't work at all?
I can't transcribe in some langauges at all.

**Comment by thewh1teagle** [comment]
I checked `whisper.cpp` with his `cli` example.
He has that issue there too but in terminal only.
If I write the output of whisper.cpp to file it works well.
I didn't checked yet how `whisper-rs` works, so I'm wondering if you capture the prints of `whisper-rs`
and if you do, why not register some callback?

**Comment by tazz4843** [comment]
We don't do anything with the string, this would be a bug in Rust's std string library, which there's essentially no chance of. As such this means it must be whisper.cpp returning an invalid UTF-8 string. We could return the raw bytes on error, but those are somewhat useless without being able to parse it unless you want to parse only up to the index where it fails.

**Comment by magnus-ISU** [comment]
UTF-8 is designed specifically to be able to recover from invalid strings, right?
<img width="570" alt="image" src="https://github.com/tazz4843/whisper-rs/assets/75949275/bd202784-f439-4db1-8956-860f4342ffa6">
You could discard whatever is invalid (seems best to me); or as [this crate](https://crates.io/crates/utf8_iter) (I think — it is dense and I didn't care to verify after glancing at the code) does, return invalid codepoints as valid UTF-8 had their prefixes been right.
```
0xxxxxxx -> great, we're back to ASCII, continue
10xxxxxx -> crap, invalid
110xxxxx -> great, back to valid input
10xxxxxx  -> end of the last char
10xxxxxx -> invalid
11110xxx -> start of 4 byte char
11xxxxxx -> invalid
11110xxx -> start of 4 byte char
10xxxxxx
10xxxxxx
10xxxxxx -> end of valid 4 byte char
```
you could still parse out of there `0xxxxxxx 110xxxxx 10xxxxxx 11110xxx 10xxxxxx 10xxxxxxx 10xxxxxx` and assuming what you had was 1 invalid codepoint and a ton of crap, it will probably be fine.

**Comment by tazz4843** [comment]
There is [`String::from_utf8_lossy`](https://doc.rust-lang.org/stable/std/string/struct.String.html#method.from_utf8_lossy) for that which does throw away information to get a valid UTF-8 string

**Comment by thewh1teagle** [comment]
I still experience this issue, I'm not sure wether it's in my control or `whisper-rs` need to be changed
https://github.com/thewh1teagle/vibe/issues/34
Can I ignore these utf-8 errors?

**Comment by tazz4843** [comment]
Remind me in a few days and I can add a function to infallibly convert.

**Comment by thewh1teagle** [comment]
Hey, just a reminder 
Many people opened issue related to that in [vibe/issues](https://github.com/thewh1teagle/vibe/issues) so I hope to solve it.
I think that it's better to receive some invalid characters than fail the whole transcription

**Comment by tazz4843** [comment]
Should be solved in f4ea0d97e48fb12f97755b4fbc2813670e177afc

**Comment by thewh1teagle** [comment]
> Should be solved in [f4ea0d9](https://github.com/tazz4843/whisper-rs/commit/f4ea0d97e48fb12f97755b4fbc2813670e177afc)

Thanks, I wasn't able to use it but it helped me understand where is the problem so I added 
[thewh1teagle/whisper-rs/commit/ee..30](https://github.com/thewh1teagle/whisper-rs/commit/ee932caf3d47d85995a587d0dd3cbd5cf932e930) and looks like it fixed the issue (and I don't even see invalid characters).
I can create PR from that if you want :)
